### PR TITLE
[AGFS fee reform] Extra changes

### PR DESCRIPTION
**Pull request by lostie**
#### > What

[Pivotal Tracker](https://www.pivotaltracker.com/story/show/156468243)

- **Daily attendance 2+** is the only Daily attendance fee [advocate final - graduate fees]
- Certification page for advocate interims should be the same as the AGFS final one
- Copy for Basic fee guidance text under the new fee reform scheme changes
- Remove help text for SAF fee under the new fee reform scheme
- Add contrary to the filter of offences search
- Disk evidence is not required for advocates interim claims
- Display PPE help text
- Order offences by alphabetic category description > alphabetic band description > alphabetic offence description
- Fix bug with adding a new defendant not displaying associated representation order details

#### > Why

All the changes implemented out of feedback on previous tickets related to _fee reform_.

#### > Deployment requirements

* New migration adding a _position_ field to **fee types** table
* **Fee types** data needs to be re-seeded:

```
SEEDS_DRY_MODE=false bundle exec rake fee_types:seed
```

### CODE REVIEW

File: lib/assets/data/fee_types.csv [hunk 1]
```diff
@@ -1,100 +1,101 @@
...
+93,Effective PCMH,IPCMH,INPCM,,FALSE,Fee::InterimFeeType,lgfs,,FALSE,
+94,Trial start,ITST,INTDT,,FALSE,Fee::InterimFeeType,lgfs,,FALSE,
+95,Retrial New solicitor,IRNS,INRNS,,FALSE,Fee::InterimFeeType,lgfs,,FALSE,
+96,Retrial start,IRST,INRST,,FALSE,Fee::InterimFeeType,lgfs,,FALSE,
+97,Disbursement only,IDISO,INDIS,,FALSE,Fee::InterimFeeType,lgfs,,FALSE,
+98,Warrant,IWARR,INWAR,,FALSE,Fee::InterimFeeType,lgfs,,FALSE,
+99,Transfer,TRANS,TRANS,,FALSE,Fee::TransferFeeType,lgfs,,FALSE,
+100,Further case management hearing,FCM,MIFCM,,TRUE,Fee::MiscFeeType,agfs;agfs_scheme_10,,FALSE,
+101,Ground rules hearing,GRH,MIGRH,,TRUE,Fee::MiscFeeType,agfs;agfs_scheme_10,,FALSE,
+102,Daily attendance fee (2+),DAT,BADAT,999,TRUE,Fee::BasicFeeType,agfs;agfs_scheme_10,,FALSE,2
```
Comment (lostie) [inline lib/assets/data/fee_types.csv:101]: New **fee type** under for scheme 10: "Daily attendance fee (2+)", that replaces all the existent "Daily attendance fee X" existent in scheme 9.

File: db/schema.rb [hunk 2]
```diff
@@ -338,6 +337,7 @@
     t.integer  "parent_id"
     t.boolean  "quantity_is_decimal", default: false
     t.string   "unique_code"
+    t.integer  "position"
```
Comment (lostie) [inline db/schema.rb:340]: This new column is meant to be used for displaying **fees/fee types** without relying on the order they've been inserted in the DB (which was the case until now). There's a potential for this to be indexed, but would like to understand a bit better what other columns are used as well for filtering.

File: app/models/offence.rb [hunk 3]
```diff
@@ -23,7 +23,7 @@ def in_scheme_nine?
     end
   end
 
-  delegate :offence_category, to: :offence_band
+  delegate :offence_category, to: :offence_band, allow_nil: true
```
Comment (colinbruce) [inline app/models/offence.rb:26]: Good catch, thanks

**Review by colinbruce — APPROVED** [review]
LGTM 👍

File: app/views/external_users/claims/basic_fees/_basic_fee_calculated_fields.html.haml [hunk 4]
```diff
@@ -4,12 +4,10 @@
...
     = f.object.description
-    - if fee.fee_type.code == 'BAF'
-      .form-hint
-        = raw t('.basic_fee_prompt_text')
-    - if fee.fee_type.code == 'SAF'
+
+    - if fee.prompt_text
       .form-hint
-        = raw t('.saf_prompt_text')
+        = fee.prompt_text
```
Comment (jsugarman) [inline app/views/external_users/claims/basic_fees/_basic_fee_calculated_fields.html.haml:10]: 👍

File: app/views/external_users/claims/offence_details/_fee_reform_fields.html.haml [hunk 5]
```diff
@@ -1,7 +1,8 @@
 - locale_scope = 'external_users.claims.offence_details'
 - selected_offence = f.object.offence
+- valid_offence = selected_offence.present? && selected_offence.offence_category.present?
 
-- if selected_offence.present?
+- if valid_offence
```
Comment (jsugarman) [inline app/views/external_users/claims/offence_details/_fee_reform_fields.html.haml:5]: boolean use `?` nitpick

**Review by jsugarman — APPROVED** [review]
👍 LGTM

**Pull request merged**